server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -897,7 +926,14 @@ private FlowInstanceConfigurer buildConfigurer(
                     nodeConfig.getAutoApproval(), approvalFlowConfig.getApprovalExpirationIntervalSeconds(),
                     nodeConfig.getExternalApprovalId());
             if (Objects.nonNull(resourceRoleId)) {
-                approvalInstance.setCandidate(StringUtils.join(flowInstanceReq.getProjectId(), ":", resourceRoleId));
+                Long candidateResourceId;
+                Optional<ResourceRoleEntity> resourceRole = resourceRoleService.findResourceRoleById(resourceRoleId);
+                if (resourceRole.isPresent() && resourceRole.get().getResourceType() == ResourceType.ODC_DATABASE) {
+                    candidateResourceId = flowInstanceReq.getDatabaseId();
+                } else {
+                    candidateResourceId = flowInstanceReq.getProjectId();
+                }
+                approvalInstance.setCandidate(StringUtils.join(candidateResourceId, ":", resourceRoleId));
```

Comment:
how can we know that which is project id and database id if we get such an expression '12:23'?

Reply:
when and where should we care about this field?